### PR TITLE
Features/suggested risk

### PR DESCRIPTION
**Pull request by joshcampbell191**
- Updated the Risk Assessment control to follow the same pattern as the Checklist control
- Added DI for Risk Assessment service
- Updated behavior of Risk Assessment controls based on the override status
- Added unit tests to cover different suggested risk scenarios
- Calculate the suggested risk after saving changes to the Risk Assessment control
  - The suggested risk should always be the highest risk appetite selected

### CODE REVIEW

File: src/Compliance.WebResources/src/controls/RiskAssessment/RiskAssessmentControl.test.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            control.save();
+            await updateSuggestedRisk;
+
+            // Assert
+            updateSuggestedRisk.should.have.been.called;
+            updateSuggestedRisk.should.have.been.calledWith("guid-test", null);
+        });
+
+        it("it should set the suggested risk to the highest selected value", async () => {
+            let riskDefinitions:(
```
Comment (pierluc-ssoucy) [inline src/Compliance.WebResources/src/controls/RiskAssessment/RiskAssessmentControl.test.ts:163]: We should extract those into helpers methods just to make the test a little more concise.

File: src/Compliance.WebResources/src/controls/RiskAssessment/RiskAssessmentControl.test.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    describe("when form is saving", () => {
+        let service: IRiskAssessmentService;
+        let xrmContext: XrmContextMock;
+
+        let documentContext: Document;
+        let placeholderElement: HTMLElement;
+
+        let control: Controls.RiskAssessmentControl;
+
+        let riskAppetites: (
```
Comment (pierluc-ssoucy) [inline src/Compliance.WebResources/src/controls/RiskAssessment/RiskAssessmentControl.test.ts:89]: We should extract those into helpers methods just to make the test a little more concise.

File: src/Compliance.WebResources/src/forms/RiskAssessmentMainForm.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            formContext.getAttribute("opc_bypassinitiatedby").setRequiredLevel(bypassSuggestedRiskAppetite ? "required" : "none");
+            formContext.getAttribute("opc_bypassrationale").setRequiredLevel(bypassSuggestedRiskAppetite ? "required" : "none");
+            formContext.getAttribute("opc_actualriskappetite").setRequiredLevel(bypassSuggestedRiskAppetite ? "required" : "none");
+
+            // Clear the fields if it's not required
+            if (!bypassSuggestedRiskAppetite) {
+                formContext.getAttribute("opc_bypassinitiatedby").setValue();
+                formContext.getAttribute("opc_bypassrationale").setValue();
+                formContext.getAttribute("opc_actualriskappetite").setValue();
+            }
```
Comment (pierluc-ssoucy) [inline src/Compliance.WebResources/src/forms/RiskAssessmentMainForm.ts:52]: this is already taken care of by the .toggle function.

File: src/Compliance.WebResources/src/services/RiskAssessmentService.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+export class RiskAssessmentService implements IRiskAssessmentService {
+    getRiskAppetites(): Promise<(opc_RiskAppetite_Fixed & { opc_riskappetiteid: string; } & { opc_name: string; } & { opc_value: number; })[]> {
+        return XrmQuery.retrieveMultiple(x => x.opc_riskappetites)
+            .select(x => [x.opc_riskappetiteid, x.opc_name, x.opc_value])
+            .filter(x => Filter.equals(x.statuscode, opc_riskappetite_statuscode.Active))
+            .orderDesc(x => x.opc_name)
+            .promise();
+    }
+
+    getRiskDefinitions(id: string): Promise<({ opc_RiskAssessmentDefinitionTemplate: opc_RiskAssessmentDefinitionTemplate_Result; } & { opc_RiskAssessmentFactorTemplate: opc_RiskAssessmentFactorTemplate_Result; } & { opc_RiskAssessmentCategory: opc_RiskAssessmentCategory_Result; } & opc_RiskAssessmentDefinition_Fixed & { opc_riskassessmentdefinitionid: string; } & { opc_riskassessmentcategory_guid: string; } & { opc_riskassessmentfactortemplate_guid: string; } & { opc_riskassessmentdefinitiontemplate_guid: string; } & { opc_isselected: boolean; })[]> {
```
Comment (pierluc-ssoucy) [inline src/Compliance.WebResources/src/services/RiskAssessmentService.ts:15]: We likely want to clean this up eventually with a tailored type, but for now, I think that's alright.

Comment (joshcampbell191) [reply src/Compliance.WebResources/src/services/RiskAssessmentService.ts:15]: Sounds good to me.

**Pull request merged**